ow_lander/scripts/all_action_trajectories.py

```diff
@@ -856,8 +855,45 @@ def deliver_sample(move_arm, robot, moveit_fk, args):
     if len(plan_b.joint_trajectory.points) == 0:  # If no plan found, send the previous plan only
         return plan_a
 
+    discard_sample_traj = cascade_plans(plan_a, plan_b)
+
+    return discard_sample_traj
+
+def deliver_sample(move_arm, robot, moveit_fk):
+    """
+    :type move_arm: class 'moveit_commander.move_group.MoveGroupCommander'
+    """
+    move_arm.set_planner_id("RRTstar")
+    robot_state = robot.get_current_state()
+    move_arm.set_start_state(robot_state)
+    goal = move_arm.get_current_pose().pose
+    goal = move_arm.get_named_target_values("arm_deliver_staging_1")
+    move_arm.set_joint_value_target(goal)
+    _, plan_a, _, _ = move_arm.plan()
+    if len(plan_a.joint_trajectory.points) == 0:  # If no plan found, abort
+        return False
+
+    cs, start_state, goal_pose = calculate_joint_state_end_pose_from_plan_arm(
+        robot, plan_a, move_arm, moveit_fk)
+    move_arm.set_start_state(cs)
+
+    goal = move_arm.get_named_target_values("arm_deliver_staging_2")
+    # move_arm.set_pose_target(goal_pose)
+    move_arm.set_joint_value_target(goal)
+    _, plan_b, _, _ = move_arm.plan()
+    if len(plan_b.joint_trajectory.points) == 0:  # If no plan found, send the previous plan only
+        return plan_a
```

Comment:
I am not in full agreement with the failure handling here .. if we fail to generate one of the activity motion plan segments, then we are unable to fulfill the request activity in its entirety. As such we abort and return false in such case.

What is the purpose of executing half of the deliver arm movement?

Reply:
Done..